### PR TITLE
chore: bump go-header, go-fraud and celestia-openrpc

### PR DESCRIPTION
**Pull request by Wondertan**
Depends on https://github.com/rollkit/celestia-openrpc/pull/43

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -189,3 +189,5 @@ replace (
 	github.com/gogo/protobuf => github.com/regen-network/protobuf v1.3.2-alpha.regen.4
 	google.golang.org/grpc => google.golang.org/grpc v1.33.2
 )
+
+replace github.com/rollkit/celestia-openrpc => ../celestia-openrpc
```
Comment (Wondertan) [inline go.mod:193]: Remove once https://github.com/rollkit/celestia-openrpc/pull/43 has merged.

File: types/header.go [hunk 2]
```diff
@@ -115,38 +104,6 @@ func (h *Header) Validate() error {
...
-
-	if !untrusted.Time().Before(time.Now().Add(maxClockDrift)) {
-		return fmt.Errorf("new header has a time from the future %v (now: %v; max clock drift: %v)",
-			untrusted.Time(),
-			time.Now(),
-			maxClockDrift)
-	}
-
-	return nil
-}
```
Comment (Wondertan) [inline types/header.go:148]: This all is now automatically done by go-header, including basic validation.

File: types/signed_header.go [hunk 3]
```diff
@@ -9,36 +9,34 @@ import (
...
 		}
 	}
 
-	// TODO: Accept non-adjacent headers until go-header implements feature to accept non-adjacent
-	if sH.Height()+1 < untrst.Height() {
-		return nil
+	if sH.Height()+1 < untrstH.Height() {
+		return &header.VerifyError{
+			Reason:      fmt.Errorf("non-adjacent headers: untrusted %d, trusted %d", untrstH.Height(), sH.Height()),
+			SoftFailure: true,
```
Comment (Wondertan) [inline types/signed_header.go:38]: Soft means header won't be discarded and will be used as a sync target.

File: types/header.go [hunk 4]
```diff
@@ -63,8 +62,8 @@ func (h *Header) ChainID() string {
 	return h.BaseHeader.ChainID
 }
 
-func (h *Header) Height() int64 {
-	return int64(h.BaseHeader.Height)
+func (h *Header) Height() uint64 {
```
Comment (Wondertan) [inline types/header.go:65]: It is now uint64, as it should be. Would be nice to make Rollkit to use uin64 everywhere as well, but that's def not part of this PR.

File: go.mod (hunk 1 above)
Comment (Manav-Aggarwal) [reply go.mod:193]: @Wondertan https://github.com/rollkit/celestia-openrpc/pull/43 is merged

File: types/header.go (hunk 4 above)
Comment (Manav-Aggarwal) [reply types/header.go:65]: PR for it: https://github.com/rollkit/rollkit/pull/1177

File: go.mod (hunk 1 above)
Comment (Wondertan) [reply go.mod:193]: Is it released?

**Review by Manav-Aggarwal — APPROVED** [review]

File: types/signed_header.go [hunk 5]
```diff
@@ -7,51 +7,62 @@ import (
...
-	if !ok {
-		// if the header type is wrong, something very bad is going on
-		// and is a programmer bug
-		panic(fmt.Errorf("%T is not of type %T", untrst, untrstH))
-	}
+func (sh *SignedHeader) Verify(untrstH *SignedHeader) error {
+	// TODO(@Wondertan):
+	//  We keep this check because of how unit tests are structured where TestVerify tests both ValidateBasic and Verify.
+	//  While the check is redundant as go-header ensures untrustH passed ValidateBasic before.
+	//  Decoupling TestVerify into TestValidateBasic and TestVerify would allow to remove this check.
```
Comment (Wondertan) [inline types/signed_header.go:41]: This may be worth to be an issue, defer to you

**Review by gupadhyaya — APPROVED** [review]

File: types/signed_header.go (hunk 5 above)
Comment (Manav-Aggarwal) [reply types/signed_header.go:41]: Made an issue here: https://github.com/rollkit/rollkit/issues/1189
